### ISSUE TITLE
Additional Messages Received Upon Reopen of Chat Window Using 15.d.backchannel-send-welcome-event

### ISSUE THREAD

**Issue by waltec888** [microsoft/BotFramework-WebChat#2805]
I am testing a webchat client that sends a backchannel event to our bot using the method proposed in this sample, [15.d.backchannel-send-welcome-event](https://github.com/microsoft/BotFramework-WebChat/tree/3aba60483eea016816b2423a0dfcf529120324a2/samples/15.d.backchannel-send-welcome-event). 

I am having some difficulty with one scenario, which results in a 'webchat/join' event being sent to the bot every time the chat window is closed/opened. 

We have a button on our page that overlays a chat window on our website. This button calls the below code when opening the chat window. 

```
function openChat(){

(async function() {
        window.WebChat.renderWebChat(
          {
            directLine: window.WebChat.createDirectLine(token), store,
   styleOptions
          },
          document.getElementById('webchat'));
        document.querySelector('#webchat > *').focus();
      })().catch(err => console.error(err));

}
```

The 'store' is defined as:

`const store = window.WebChat.createStore(
      	{},
      	({ dispatch }) => next => action => {
          console.log(">>> HTML DISPATCH action: " + action.type);
          if (action.type == 'DIRECT_LINE/CONNECT_FULFILLED') {
            dispatch({
              type: 'WEB_CHAT/SEND_EVENT',
              payload: {
                name: 'webchat/join',
              }
            });
          }
          return next(action);
        }
      );`

I have tested calling the window.WebChat.renderWebChat method without passing the store object, but the window.WebChat object seems to be retaining the store that was passed in the initial call to window.WebChat.renderWebChat. 

Is there a way to clear out the store or remove the dispatch if I am simply re-opening a window (overlay) that was previously closed/hidden? OR Am I handling how the chat window overlay is closing/re-opening incorrectly? I haven't had success in finding documentation aside from the sample I found originally.

**Comment by waltec888** [comment]
I am not sure if there is a recommended approach for how to ensure a webchat/join event is only sent once; however, I did find a way to modify our code to ensure the action is only dispatched for the initial opening of the chat overlay. 

If this differs from the recommended way, or there is a better way to achieve this functionality, please let me know!

`const store = window.WebChat.createStore(
      	{},
      	({ dispatch }) => next => action => {
          console.log(">>> HTML DISPATCH action: " + action.type);
          if (action.type == 'DIRECT_LINE/CONNECT_FULFILLED') {
		sendEvent();
          }
          return next(action);
        }
      );

var newT=false;
      
function openChat(newToken){

newT = newToken;
        window.WebChat.renderWebChat(
          {
            directLine: window.WebChat.createDirectLine(token), store,
   styleOptions
          },
          document.getElementById('webchat'));

        document.querySelector('#webchat > *').focus();
}

function sendEvent(){
if (newT){
store.dispatch({
           type: 'WEB_CHAT/SEND_EVENT',
           payload: { name: 'webchat/join' }
         });
}
}`

**Comment by tdurnford** [comment]
@waltec888

There are currently two ways to trigger a welcome message in Web Chat. The first option is to dispatch a welcome event when DirectLineJs connects to the bot. As  you noticed, this triggers the welcome message every time Web Chat renders. https://github.com/microsoft/BotFramework-WebChat/issues/2750 is currently discussing changing this behavior.

The second option is to generate a Direct Line token with a user id. Following this approach will send the conversation update event -  which should only be invoked when the conversation is initialized -  for the user before the user messages the bot. This will allow you to send a traditional welcome message from the bot. For more details take a look at the  [Web Chat Welcome Message](https://github.com/microsoft/BotFramework-WebChat/blob/master/WELCOME_MESSAGE.md) documentation.

**Comment by waltec888** [comment]
It looks like this behavior has been fixed last night. I see it listed in the [changelog](https://github.com/microsoft/BotFramework-WebChat/blob/master/CHANGELOG.md) (2nd bullet under Fixed), but I do not see that there has been a [release](https://github.com/microsoft/BotFramework-WebChat/releases). 

I am new to github, and want to make sure I'm looking in the right place. Is 4.7.1 still the latest release? If not, did I just look at the releases page before it was updated?

**Comment by tdurnford** [comment]
@waltec888 We released v4.7.1 of Web Chat in mid-December, and our next major release - v4.8.0 - should be released some time around March. 

It looks like the change you are referring to was made to the Minimizable Web Chat Sample. Our samples are updated on a regular basis, and you are always free to pull the latest changes from our GitHub Repo.

**Comment by waltec888** [comment]
That makes sense. Thank you for clarifying!! :) There must have been a change on our side to how we are handling minimize that is now causing minimize/maximize to work properly.
